x-pack/platform/plugins/shared/streams/server/lib/streams/state_management/execution_plan/execution_plan.ts

```diff
@@ -93,7 +96,50 @@ export class ExecutionPlan {
     return this.actionsByType;
   }
 
+  async validatePermissions() {
+    const { actionsByType } = this;
+
+    const requiredPermissions = getRequiredPermissionsForActions(actionsByType);
+
+    // Check if we have any permissions to validate
+    if (
+      requiredPermissions.cluster.length === 0 &&
+      Object.keys(requiredPermissions.index).length === 0
+    ) {
+      return true;
+    }
```

Comment:
Are there any actions that don't require any privileges? Or does this indicate that there are no actions?